package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -1,246 +1,324 @@
-    - Each child of a node could be a data node, a task node, a parameters node or
-    another modular pipeline node.
-    - There is a designated root node with a __root__ ID.
-
-    With this representation, a folder-like render of the tree is simply a recursive in-order
-    tree traversal. To improve the performance on the client, we perform the conversion between
-    these two representations on the backend by extracting the modular pipeline from a Kedro project
-    and adding it to this repository.
-    """
+    """Repository for the set of modular pipelines in a registered pipeline."""
 
     def __init__(self):
-        # The tree representation of the tree.
-        # Example:
-        # {
-        #   "__root__": ModularPipelineNode(
-        #       id="__root__",
-        #       children=["data_science", "data_engineering"]
-        #   ),
-        #   "data_science": ModularPipelineNode(id="data_science", children=[]),
-        #   "data_engineering": ModularPipelineNode(id="data_science", children=[]),
-        # }
+        """
+        Initialize the ModularPipelinesRepository with an
+        empty tree and have a node to modular pipeline mapping.
+
+        The tree contains a root modular pipeline node.
+        """
         self.tree: Dict[str, ModularPipelineNode] = {
             ROOT_MODULAR_PIPELINE_ID: GraphNode.create_modular_pipeline_node(
                 ROOT_MODULAR_PIPELINE_ID
             )
         }
+        self.node_mod_pipeline_map: Dict[
+            str, Set[str]
+        ] = {}  # Updated to map node_id to a set of modular_pipeline_ids
+
+        self.parameters = set()
+
+    def populate_tree(self, pipeline: KedroPipeline):
+        """
+        Add inputs, outputs, and children to each modular
+        pipeline node based on the given Kedro pipeline.
+
+        Args:
+            pipeline (KedroPipeline): The Kedro pipeline to populate the tree from.
+        """
+        namespaces = sorted(
+            {node.namespace for node in pipeline.nodes if node.namespace}
+        )
+
+        modular_pipeline_ids = {
+            id for ns in namespaces for id in self._explode_namespace(ns)
+        }
+
+        for modular_pipeline_id in sorted(modular_pipeline_ids):
+            self.get_or_create_modular_pipeline(modular_pipeline_id)
+
+            sub_pipeline = pipeline.only_nodes_with_namespace(modular_pipeline_id)
+            rest_of_the_pipeline = pipeline - sub_pipeline
+
+            free_inputs = sub_pipeline.inputs()
+            free_outputs = sub_pipeline.outputs() | (
+                rest_of_the_pipeline.inputs() & sub_pipeline.all_outputs()
+            )
+
+            self.add_inputs(modular_pipeline_id, free_inputs)
+            self.add_outputs(modular_pipeline_id, free_outputs)
+            self.add_children(modular_pipeline_id, sub_pipeline.nodes)
+
+    def _explode_namespace(self, nested_namespace: str) -> List[str]:

```

Comment:
Why does this need to take `self`? This doesn't need an instance of `ModularPipelinesRepository` so it can just be a static method or it can be defined outside the `ModularPipelinesRepository` class altogether.

Reply:
Yes, converted to static method. Thank you